org.mixedrealitytoolkit.uxcore/StateVisualizer/StateVisualizer.cs

```diff
@@ -301,7 +301,11 @@ private void Update()
             }
 
             // If we're asleep, quit early.
-            if (!animator.enabled) { return; }
+            if (!animator.enabled) 
+            {
+                enabled = false;
```

Comment:
Can you confirm this will be re-enabled once the animator is enabled?

Reply:
Hi  @AMollis, Not sure I understand your question, but this is what I know:

The `StateVisualizer` controls the animator. After `StateVisualizer` wakes up from any of these events:
```c#
unsubscribeActions.Add(Subscribe(interactable.hoverEntered, WakeUp));
unsubscribeActions.Add(Subscribe(interactable.hoverExited, WakeUp));
unsubscribeActions.Add(Subscribe(interactable.selectEntered, WakeUp));
unsubscribeActions.Add(Subscribe(interactable.selectExited, WakeUp));
unsubscribeActions.Add(Subscribe(interactable.IsToggled.OnEntered, WakeUp));
unsubscribeActions.Add(Subscribe(interactable.IsToggled.OnExited, WakeUp));
unsubscribeActions.Add(Subscribe(interactable.OnEnabled, WakeUp));
unsubscribeActions.Add(Subscribe(interactable.OnDisabled, WakeUp));
```

For context, this is the `WakeUp` method:

```c#
private void WakeUp()
{
    enabled = true;
    sleepTimer = keepAliveTime;
}
```

It will re-evaluate if a change happened (in the Update loop) and then re-enable the animator.